src/Event.cc

```diff
@@ -132,16 +119,13 @@ void EventMgr::Drain() {
     // that expect the old behavior to trigger something quickly.
 
     for ( int round = 0; head && round < 2; round++ ) {
-        Event* current = head;
+        current = head;
         head = nullptr;
         tail = nullptr;
 
         while ( current ) {
             Event* next = current->NextEvent();
 
-            current_src = current->Source();
-            current_aid = current->Analyzer();
-            current_ts = current->Time();
             current->Dispatch();
             Unref(current);
```

Comment:
Minor: I find this usage of the non-local `current` a bit hard to read/follow. Could instead leave the iteration variable as a local (say `Event *ev`), and then call `Dispatch(ev)` to do the dispatch and delete.

Reply:
Thanks - yes, I'll revert. I'll check on the ASAN failure, too :sweat:

Reply:
He, this is exactly where the ASAN bug was. Thanks!